docs/features/software-catalog/extending-the-model.md

```diff
@@ -585,6 +585,45 @@ export class FoobarEntitiesProcessor implements CatalogProcessor {
 }
 ```
 
+### New Backend
+
+You should generally create a new module to hold your new processor. You can create a new backend module using the `backstage-cli create` command and selecting `backend-module` option. To create a new module, you need a plugin ID and a module ID. We'll be using `catalog` as our plugin ID since our module is adding/updating catalog functionality. For module ID, we'll use `foobar`, but this should match the ID of whatever your plugin that's integrating with the catalog is, for example, AWS would be `aws`, Backstage Search would be `search`, etc.
```

Comment:
The thing about creating a module, should that go in the previous section instead? We've already told the user to create all of the processor code and it has to go somewhere. Perhaps it's even best to merge the two sections entirely. The processor could go right next to the module declaration honestly, and not even be exported.

Nit: Search as an example is perhaps a bit unexpected; not likely that the catalog will want to talk to search (it goes the other way around typically)

Reply:
Good call, I updated this. The docs above are a bit confusing, the `FoobarEntitiesProcessor` is implied to sit in the `-common` package, but that's not super clear in the docs. I updated it to hopefully be a bit more helpful, but there's definitely adjustments to be made. 

I also tried to follow the new `--old` pattern, let me know if my usage makes sense?